app/controllers/posts_controller.rb

```diff
@@ -160,6 +160,67 @@ def show
 
   def edit; end
 
+  # Attempts to update a given post
+  # @param post [Post] post the user is attempting to update
+  # @param user [User] user attempting to update the post
+  # @param edit_post_params [ActionController::Parameters] edit parameters
+  # @param body_rendered [String] new post body
+  # @return [Boolean] status of the operation
+  def do_update(post, user, edit_post_params, body_rendered)
+    post.update(edit_post_params.merge(body: body_rendered,
+                                       last_edited_at: DateTime.now,
+                                       last_edited_by: user,
+                                       last_activity: DateTime.now,
+                                       last_activity_by: user))
+  end
```

Comment:
question: what's the benefit of having this method over just calling `post.update` manually?

Reply:
Frankly, not much at this point. Initially, I expected both `do_update` and `do_update_network` to include history updates, but it turned out to not be desirable. Right now it's like that mostly for consistency with `do_update_network` that _does_ do more heavy lifting + to ease understanding of what the `update` method does. If that's fine with you, I'd keep it that way as its complexity is likely to grow?